File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int mx=INT_MIN;
        int sm=0;
        int n=nums.size();
        for(int i=0;i<n;i++ )
        {
            sm+=nums[i];
             mx=max(sm,mx);
            if(sm<0)
            {
                sm=0;
            }
           
        }
        return mx;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let mx = Number.MIN_SAFE_INTEGER;
     let sm = 0;
    let n=nums.length;
    for(let i=0;i<n;i++)
        {
            sm+=nums[i];
            mx=Math.max(sm,mx);
            if(sm<0)
                {
                    sm=0;
                }
        }
    return mx;
};